gen_initramfs.sh

```diff
@@ -1284,6 +1285,137 @@ append_splash() {
+	fi
+	if [ -z "${PLYMOUTH_THEME}" -a -e /etc/plymouth/plymouthd.conf ]
+	then
+		PLYMOUTH_THEME=$(plymouth-set-default-theme) || gen_die "Failed to set default plymouth theme!"
+	fi
+	if [ -z "${PLYMOUTH_THEME}" ]
+	then
+		PLYMOUTH_THEME=text
+	fi
+
+	print_info 1 "$(get_indent 1)>> Installing plymouth [ using the '${PLYMOUTH_THEME}' theme ]..."
+
+	/usr/libexec/plymouth/plymouth-populate-initrd -t "${TDIR}" \
+		|| gen_die "Failed to build plymouth cpio archive!"
+
+	# can probably get rid of this; depends if plymouth was built with static libs
+	# rm -f "${TDIR}"/lib*/{ld*,libc*,libz*} \
+		# || gen_die "Failed to clean up plymouth cpio archive!"
+
+	ln -sf "${PLYMOUTH_THEME}/${PLYMOUTH_THEME}.plymouth" "${TDIR}/usr/share/plymouth/themes/default.plymouth" \
+		|| gen_die "Failed to set the default plymouth theme!"
+
+	# this might be better located in append_drm (?)
+	mkdir -p "${TDIR}"/usr/lib/udev/rules.d || gen_die "Failed to create '${TDIR}/usr/lib/udev/rules.d'!"
+	cp -aL /lib/udev/rules.d/70-uaccess.rules "${TDIR}/usr/lib/udev/rules.d" || gen_die "Failed to copy '70-uaccess.rules'!"
+	cp -aL /lib/udev/rules.d/71-seat.rules "${TDIR}/usr/lib/udev/rules.d" || gen_die "Failed to copy '71-seat.rules'!"
+
+	# clean up
+	cd "${TDIR}" || gen_die "Failed to chdir to '${TDIR}'!"
+	log_future_cpio_content
+	find . -print0 | "${CPIO_COMMAND}" ${CPIO_ARGS} --append -F "${CPIO_ARCHIVE}" \
+		|| gen_die "Failed to append ${PN} to cpio!"
+
+	cd "${TEMP}" || die "Failed to chdir to '${TEMP}'!"
+	if isTrue "${CLEANUP}"
+	then
+		rm -rf "${TDIR}"
+	fi
+}
+
+append_drm() {
+	local PN=drm
+	local TDIR="${TEMP}/initramfs-${PN}-${KV}-temp"
+	if [ -d "${TDIR}" ]
+	then
+		rm -r "${TDIR}" || gen_die "Failed to clean out existing '${TDIR}'!"
+	fi
+
+	mkdir "${TDIR}" || gen_die "Failed to create '${TDIR}'!"
+	cd "${TDIR}" || gen_die "Failed to chdir to '${TDIR}'!"
+
+	print_info 1 "$(get_indent 1)>> Appending drm drivers..."
+
+	# mkdir -p "${TDIR}/lib/modules/${KV}"
+
+	local drm_path="/lib/modules/${KV}/kernel/drivers/gpu/drm"
+	local modules
+	if [ -d "${drm_path}" ]
+	then
+		modules=$(strip_mod_paths $(find "${drm_path}" -name "*.ko"))
+	else
+		print_info 1 "$(get_indent 2)Warning :: no drm modules in drivers/gpu/drm..."
+	fi
+
+	rm -f "${TEMP}/moddeps" || gen_die "Failed to clear old moddeps!"
+	gen_deps ${modules}
+	if [ -f "${TEMP}/moddeps" ]
+	then
+		modules=$(cat "${TEMP}/moddeps" | sort | uniq)
+	else
+		print_info 1 "$(get_indent 2)Warning :: module dependencies not generated..."
+	fi
+
+	local mod i fws fw
+	for i in ${modules}
+	do
+		mod=$(find "/lib/modules/${KV}" -name "${i}.ko" 2>/dev/null| head -n 1)
+		if [ -z "${mod}" ]
+		then
+			print_info 1 "$(get_indent 2)Warning :: ${i}.ko not found; skipping..."
+			continue
+		fi
+
+		print_info 1 "$(get_indent 2)>> Copying ${mod}..."
+		cp -ax --parents "${mod}" "${TDIR}" || gen_die "Failed to copy '${mod}'!"
```

Comment:
Why not use the normal modules functionality? This is instead causing the module to be embedded into the initramfs twice potentially (once in the drm fragment, and again in the module fragment)

Reply:
Back when I originally wrote this I didn't understand the append_modules function properly (and honestly I still don't).  So if I understand you correctly, I don't actually need any of the module/firmware appending code in append_drm() at all since append_modules() will pull it all in automatically?

Reply:
I feel really silly now; I did some testing and the existing --firmware and --all-ramdisk-modules options are sufficient for plymouth to function.  My append_drm function apparently didn't anything at all either, my kernel size didn't change at all after removing it.

So, that resolves these change requests and also makes building initramfs with plymouth significantly faster.